sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -69,14 +69,15 @@ void gnrc_lorawan_mlme_confirm(gnrc_lorawan_t *mac, mlme_confirm_t *confirm)
 
 static inline void _set_be_addr(gnrc_lorawan_t *mac, uint8_t *be_addr)
 {
-    uint32_t tmp = *((uint32_t*) be_addr);
-    tmp = byteorder_swapl(tmp);
+    uint32_t tmp = byteorder_bebuftohl(be_addr);
+    le_uint32_t dev_addr = byteorder_btoll(byteorder_htonl(tmp));
```

Comment:
My brain hurts - so this does BE -> LE -> BE -> LE?!

Why not just
```suggestion
    le_uint32_t dev_addr;
    dev_addr.u32 = byteorder_bebuftohl(be_addr);
```

Reply:
You read it wrong. It is not BE -> LE -> BE -> LE. It is BE -> host byte order -> LE. Your proposal assumes that host byte order is always LE.

Reply:
I would much rather have `byteorder_bebuftohl()` do the conversion to host byte order

Reply:
I've always missed these functions. All LoRaWAN frames are in Little Endian.
Maybe it's time to add those so we avoid these dirty conversions?

Reply:
But... that's what happening here...

Reply:
> I would much rather have byteorder_bebuftohl() do the conversion to host byte order

The problem is, we need a LE buffer in the end (the DevAddr buffer is copied directly to the LoRaWAN packet).
I think we need something like a `byteorder_lebuftohl` or `byteorder_htoll`

Reply:
Now _I'm_ confused. How does `byteorder_bebuftohl()`, a big endian to host byte order function, help with little endian?

Reply:
(and it's not made better by Github not syncing properly again...)

Reply:
> Now I'm confused. How does byteorder_bebuftohl(), a big endian to host byte order function, help with little endian?

It only helps assuming all these byteorder functions require uint32_t numbers, so it's the entry point. The only things I need here is to reverse the input buffer regardless of the host.

Reply:
Yeah, I got that, I was just thinking you were talking about using `byteorder_bebuftohl()` to get to LE. However, I would rather not postpone this for another API update in `byteorder`. There are other places (IEEE 802.15.4) that could benefit from LE-to-H and vice-versa as well.